packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -159,6 +159,12 @@ export const $lib = createTypeSpecLibrary({
         default: `Cannot pass an empty value to the @clientName decorator`,
       },
     },
+    "override-method-parameters-mismatch": {
+      severity: "error",
+      messages: {
+        default: paramMessage`Method "${"methodName"}" is not directly referencing the same parameters as in the original operation. There needs to be a 1-1 mapping.`,
```

Comment:
Can we also provide more details in the diagnostics on which params are missing/new in the overridden method?